CarboPlanner/src/views/HomeScreen.tsx

```diff
@@ -50,70 +51,78 @@ export default function HomeScreen(){
+    
       return (
-        <List name={uuidObject.name} imageURI={uuidObject.icon} uuidKey={uuidKey}>
+        <List key={uuidKey} name={meal.name} imageURI={meal.icon} uuidKey={uuidKey}>
           <HorizontalLine />
-          {mealsArray.map((meal, index) => (
-            <ListItem
-              key={index.toString()} // Ensure each item has a unique key
-              weight={meal.weight}
-              name={meal.name}
-              calories={meal.calories}
-              protein={meal.protein}
-              carbs="32" // Assuming the carbs value is fixed
-              fat={meal.fat}
-            />
-          ))}
+          {Object.values(mealsArray).map((mealInfo, index) => {
+            // Extract the UUID from the mealInfo object
+            const mealUuid = Object.keys(mealsArray)[index]; 
+            return (
+              <ListItem
+                key={`${uuidKey}_${index}`} // Ensure each item has a unique key
+                weight={mealInfo.weight}
+                name={mealInfo.name}
+                calories={mealInfo.calories}
+                protein={mealInfo.protein}
+                carbs={mealInfo.carbs}
+                fat={mealInfo.fat}
+              />
+            );
+          })}
         </List>
-      );
-    };
+      )};
+    
+    
+    
+    
+
 
   return (
     <View style={styles.container}>
-      <View style={{flexDirection: 'row', alignItems: 'center'}}>
-        <Text style={styles.header_text}>Today</Text>
-        <TouchableOpacity onPress={goToCalendar}>
-          <View style={styles.calendarIconContainer}>
-            <Image source={require("../../assets/func-icon/calendar.png")} style={styles.calendarIcon} resizeMode="contain" />
-          </View>
-        </TouchableOpacity>
-      </View>
+    <View style={{flexDirection: 'row', alignItems: 'center'}}>
+      <Text style={styles.header_text}>Today</Text>
+      <TouchableOpacity onPress={goToCalendar}>
+        <View style={styles.calendarIconContainer}>
+          <Image source={require("../../assets/func-icon/calendar.png")} style={styles.calendarIcon} resizeMode="contain" />
+        </View>
+      </TouchableOpacity>
+    </View>
 
       <Card>
         <View style={styles.row}>
           <CircularSlider value={1200} max={2000}/>
         </View>
         <View style={styles.row}>
-            <MacroProgressBar name={"Carbs"} value={50} max={210} />
+            <MacroProgressBar name={"Carbs"} value={0.5} max={210} />
           
-            <MacroProgressBar name={"Protein"} value={100} max={180} />
+            <MacroProgressBar name={"Protein"} value={1} max={180} />
           
-            <MacroProgressBar name={"Fat"} value={80} max={200} />
+            <MacroProgressBar name={"Fat"} value={0.2} max={200} />
         </View>
       </Card>
-
       <ScrollView style={{paddingBottom: 10}}>
-        <Card customStyle={{maxHeight: "100%"}}>
+        <Card customStyle={{maxHeight: "44%"}}>
```

Comment:
Changed in another branch
```suggestion
        <Card customStyle={{maxHeight: "42%"}}>
```